datalad/cmdline/main.py

```diff
@@ -377,13 +377,9 @@ def main(args=None):
                 if exc.msg:
                     os.write(2, exc.msg.encode() if isinstance(exc.msg, text_type) else exc.msg)
                 if exc.stdout:
-                    os.write(1, exc.stdout.encode()) \
-                        if hasattr(exc.stdout, 'encode')  \
-                        else os.write(1, exc.stdout)
+                    os.write(1, exc.stdout.encode() if isinstance(exc.stdout, text_type) else exc.stdout)
```

Comment:
Hm... Could you describe for my education how it was failing before and now doesn't? (Since there is no test)

Reply:
The message is in the top-level comment. This is all I got.

Reply:
And obviously after the change I could successfully run that command. Sorry, this is just a quick fix to get it working, no time for further forensics now.